src/storage/mod.rs

```diff
@@ -159,14 +161,16 @@ impl fmt::Debug for Command {
 }
 
 pub struct Storage {
+    engine: Arc<Box<Engine>>,
     tx: Sender<Message>,
     thread: JoinHandle<Result<()>>,
 }
 
 impl Storage {
     pub fn from_engine(engine: Box<Engine>) -> Result<Storage> {
         let desc = format!("{:?}", engine);
-        let mut scheduler = Scheduler::new(engine);
+        let shared = Arc::new(engine);
```

Comment:
Arc is used for sharing the engine between threads here. The name 'shared' is inspired by the Arc document: https://doc.rust-lang.org/stable/std/sync/struct.Arc.html#examples